##### № 27623 Апробация 04.03.26 (Уровень: Базовый)

На предприятии каждой изготовленной детали присваивают серийный номер, содержащий десятичные цифры, 26 латинских букв (без учёта регистра) и символы из 8164-символьного специального алфавита. В базе данных для хранения каждого серийного номера отведено одинаковое и минимально возможное число байт. При этом используется посимвольное кодирование серийных номеров, все символы кодируются одинаковым и минимально возможным числом бит. Известно, что для хранения 835 серийных номеров требуется более 156 Кбайт памяти. Определите минимально возможную длину серийного номера. В ответе запишите только целое число.


In [4]:
from math import *
glubina = ceil(log2(26 + 10 + 8164))
dlina = 192 / glubina 
print(dlina * 8)

109.71428571428571


##### № 27766 Апробация 04.03.26 (Уровень: Базовый)


На предприятии каждой изготовленной детали присваивают серийный номер, содержащий десятичные цифры, 26 латинских букв (без учёта регистра) и символы из 34-символьного специального алфавита. В базе данных для хранения каждого серийного номера отведено одинаковое и минимально возможное число байт. При этом используется посимвольное кодирование серийных номеров, все символы кодируются одинаковым и минимально возможным числом бит. Известно, что для хранения 1142 серийных номеров требуется более 305 Кбайт памяти. Определите минимально возможную длину серийного номера. В ответе запишите только целое число.

In [17]:
from math import *
glubina = ceil(log2(26 + 34 + 10))
bytes_one = 274
for i in range(1,1000):
    if ceil((i * glubina)/8) >= bytes_one:
        print(i)
        break

313


##### № 24040 (Уровень: Базовый)

На предприятии каждой изготовленной детали присваивают серийный номер, содержащий десятичные цифры, 52 латинские буквы (с учётом регистра) и символы из 1989-символьного специального алфавита. В базе данных для хранения каждого серийного номера отведено одинаковое и минимально возможное число байт. При этом используется посимвольное кодирование серийных номеров, все символы кодируются одинаковым и минимально возможным числом бит. Известно, что для хранения 836 серийных номеров отведено не более 639 Кбайт памяти. Определите максимально возможную длину серийного номера. В ответе запишите только целое число.

In [23]:
from math import *
glubina = ceil(log2(52 + 1989 + 10))
bytes_one = 782
for i in range(1,10000):
    if ceil(i * glubina)/8 <= bytes_one:
        if i == 521:
            print(i)

521


##### № 24035 (Уровень: Базовый)


На предприятии каждой изготовленной детали присваивают серийный номер, состоящий из 2783 символов. В базе данных каждый серийный номер занимает одинаковое и минимально возможное целое число байт. При этом используется посимвольное кодирование серийных номеров, все символы кодируются одинаковым и минимально возможным целым числом бит.
Известно, что для хранения 62 784 серийных номеров требуется не менее 356 Мбайт памяти. Определите минимально возможную мощность алфавита, используемого для записи серийных номеров. В ответе запишите только целое число.

In [25]:
from math import *
dlina = 2783
for i in range(1,1000):
    if (62784 * ceil(i * dlina /8)) >= (356 * 1024 * 1024):
        print(i)
        break
print(2**17)

18
131072


##### № 23877 (Уровень: Базовый)


На предприятии каждой изготовленной детали присваивают серийный номер, содержащий десятичные цифры и символы из 2040-символьного специального алфавита. В базе данных каждый серийный номер занимает одинаковое и минимально возможное число байт. При этом используется посимвольное кодирование серийных номеров, все символы кодируются одинаковым и минимально возможным числом бит. Известно, что для хранения 87 000 серийных номеров требуется более 10 Мбайт памяти. Определите минимально возможную длину серийного номера.

In [27]:
from math import *
glubina = ceil(log2(2040 + 10))
bytes_one = 121
for i in range(1,1000):
    if ceil(i * glubina/8) >= bytes_one:
        print(i)
        break

81


##### № 23875 (Уровень: Базовый)

На предприятии каждой изготовленной детали присваивают серийный номер, содержащий десятичные цифры, 52 латинские буквы (с учётом регистра) и символы из 50-символьного специального алфавита. В базе данных каждый серийный номер занимает одинаковое и минимально возможное число байт. При этом используется посимвольное кодирование серийных номеров, все символы кодируются одинаковым и минимально возможным числом бит. Известно, что для хранения 800177 серийных номеров требуется более 35 Мбайт памяти. Определите минимально допустимую длину серийного номера. В ответе запишите только целое число.

In [29]:
from math import *
glubina = ceil(log2(52 + 10 + 50))
bytes_one = 46
for i in range(1,10000):
    if ceil(i * glubina /8) >= bytes_one:
        print(i)
        break

52


##### № 23874 (Уровень: Базовый)


На предприятии каждой изготовленной детали присваивают серийный номер, состоящий из 576 символов. В базе данных каждый серийный номер занимает одинаковое и минимально возможное целое число байт. При этом используется посимвольное кодирование серийных номеров, все символы кодируются одинаковым и минимально возможным целым числом бит.
Известно, что для хранения  85 627 серийных номеров требуется не менее 54 Мбайт памяти. Определите минимально возможную мощность алфавита, используемого для записи серийных номеров. В ответе запишите только целое число.

In [32]:
from math import *
dlina = 576
for i in range(1,1000):
    if (85627 * ceil(i * dlina /8)) >= (54* 1024 * 1024):
        print(i)
        break
print(2**9)

10
512


##### № 29964 Апробация 14.05.26 (Уровень: Базовый)

На предприятии каждой изготовленной детали присваивают серийный номер, содержащий десятичные цифры и символы из 62-символьного специального алфавита. В базе данных каждый серийный номер занимает одинаковое и минимально возможное число байт. При этом используется посимвольное кодирование серийных номеров, все символы кодируются одинаковым и минимально возможным числом бит. Известно, что 5 895 222 серийных номера занимают более 23 Мбайт памяти. Определите минимально возможную длину серийного номера.

In [1]:
from math import *

for x in range(1, 1000):
    alf = 10 + 62
    lenc = x
    ms = ceil(log2(alf))
    v = ceil(ms * lenc / 8)
    if v * 5895222 > 23 * 1024 * 1024:
        print(x)
        break

5


Ответ: 5

##### № 29343 Открытый вариант 2026 (Уровень: Базовый)

При регистрации в компьютерной системе каждому объекту присваивается идентификатор, состоящий из 257 символов и содержащий только цифры семнадцатеричной системы счисления и символы из 4080-символьного специального алфавита. В базе данных для хранения каждого идентификатора отведено одинаковое и минимально возможное целое число байт. При этом используется посимвольное кодирование идентификаторов, все символы кодируются одинаковым и минимально возможным количеством бит. Определите объём памяти (в Мбайт), необходимый для хранения 8 388 608 идентификаторов. В ответе запишите только целое число – количество Мбайт.

In [2]:
from math import *

# for x in range(1, 1000):
alf = 4080 + 17
lenc = 257
bit = ceil(log2(alf))
v = ceil(lenc * bit / 8)
print(v * 8388608 / 1024 / 1024)  

3344.0


Ответ: 3344

##### № 28932 ЕГКР 18.04.26 (Уровень: Базовый)

На предприятии каждой изготовленной детали присваивают серийный номер, состоящий из 65 символов. В базе данных каждый серийный номер занимает одинаковое и минимально возможное целое число байт. При этом используется посимвольное кодирование серийных номеров, все символы кодируются одинаковым и минимально возможным числом бит. Известно, что для хранения 131 072 серийных номеров отведено менее 9 Мбайт памяти. Определите максимально возможную мощность алфавита, используемого для записи серийных номеров. В ответе запишите только целое число.

In [ ]:
from math import *

for x in range(1000, 1, -1):
    alf = x
    lenc = 65
    bit = ceil(log2(alf))
    v = ceil(lenc * bit / 8)
    if v * 131072 < 9 * 1024 * 1024:
        print(x)
        break

256


Ответ: 256